webhook.go

```diff
@@ -7,30 +7,62 @@ import (
 
 // A Webhook represents an EasyPost webhook callback URL.
 type Webhook struct {
-	ID         string     `json:"id,omitempty"`
-	Object     string     `json:"object,omitempty"`
-	Mode       string     `json:"mode,omitempty"`
-	URL        string     `json:"url,omitempty"`
-	DisabledAt *time.Time `json:"disabled_at,omitempty"`
+	ID            string     `json:"id,omitempty"`
+	Object        string     `json:"object,omitempty"`
+	Mode          string     `json:"mode,omitempty"`
+	URL           string     `json:"url,omitempty"`
+	DisabledAt    *time.Time `json:"disabled_at,omitempty"`
+	WebhookSecret string     `json:"webhook_secret,omitempty"`
 }
 
-type createWebhookRequest struct {
+// CreateUpdateWebhookOptions is used to specify parameters for creating and updating EasyPost webhooks.
+type CreateUpdateWebhookOptions struct {
+	URL           string `json:"url,omitempty"`
+	WebhookSecret string `json:"webhook_secret,omitempty"`
+}
+
+// createUpdateWebhookRequest is the request struct for creating and updating webhooks (internal)
+type createUpdateWebhookRequest struct {
 	Webhook *Webhook `json:"webhook,omitempty"`
 }
 
+// listWebhooksResult is the response struct of listing webhooks (internal)
 type listWebhooksResult struct {
 	Webhooks *[]*Webhook `json:"webhooks,omitempty"`
 }
 
+func (c *Client) composeCreateUpdateWebhookRequest(data *CreateUpdateWebhookOptions) *createUpdateWebhookRequest {
+	return &createUpdateWebhookRequest{
+		Webhook: &Webhook{
+			URL:           data.URL,
+			WebhookSecret: data.WebhookSecret,
+		},
+	}
+}
+
 // CreateWebhook creates a new webhook with the given URL.
+// Deprecated: Use CreateWebhookWithDetails instead.
 func (c *Client) CreateWebhook(u string) (out *Webhook, err error) {
-	return c.CreateWebhookWithContext(context.Background(), u)
+	return c.CreateWebhookWithDetails(&CreateUpdateWebhookOptions{URL: u})
+}
+
+// CreateWebhookWithDetails creates a new webhook with the provided details.
+func (c *Client) CreateWebhookWithDetails(data *CreateUpdateWebhookOptions) (out *Webhook, err error) {
```

Comment:
**NOTE**

The idea being, since the name `CreateWebhook` is unfortunately taken, once it's removed, we deprecate `CreateWebhookWithDetails` and rename it back to `CreateWebhook` (little Tower of Hanoi)

Reply:
I hate it, but yeah.